### PR TITLE
Added script to generate table of content

### LINKED ISSUE #1528: The `INSTALLATION.md` file needs a newly generated table of contents
**Is your feature request related to a problem? Please describe.**
1. There are many missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Many of the existing TOC entries are under an inappropriate parent heading

This increases the risk of installation instructions being overlooked.

**Describe the solution you'd like**
1. Add the missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Adjust the resulting TOC entries under the appropriate parent headings
1. Ideally, the auto-generation of the TOC should be automatic as part of the `git` process

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- Consider using a markdown TOC autogenerator

**Additional context**
- N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Veer0x1**
**What kind of change does this PR introduce?**
This PR adds a script `generate:toc` to automatically generate table of contents for all the makrdown files in repository, for this command to work, there should we a comment of `<!-- toc -->` and `<!-- tocstop -->`. The table of content will be place under this comments. If this comments aren't present in any markdown file then there will be no change in those files.
I have placed this comments in all the neccessary markdown files. This done via a `markdow-toc` npm package. I have also added instruction on how to use this script in `CONTRIBUTING.md` file

**Issue Number:**

Fixes #1528 

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Review by palisadoes — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
     "generate:ssl-private-key": "openssl genrsa -out ./key.pem",
     "import:sample-data": "ts-node ./src/utilities/loadSampleData.ts",
     "import:sample-data:prod": "node ./build/utilities/loadSampleData.js",
-    "gen:schema": "graphql-inspector introspect ./src/typeDefs/**/**/*.ts --write ./schema.graphql "
+    "gen:schema": "graphql-inspector introspect ./src/typeDefs/**/**/*.ts --write ./schema.graphql ",
+    "generate:toc": "find . -maxdepth 1 -name '*.md' -exec markdown-toc -i {} \\;"
```
Comment (palisadoes) [inline package.json:26]: Will this now be run automatically as a githook for commits?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: package.json (hunk 1 above)
Comment (Veer0x1) [reply package.json:26]: Currently it is not. But if you say then I'll add.

Comment (palisadoes) [reply package.json:26]: Now that this option has been created, adding it to a githook will be a welcome addition.

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

**Comment by Veer0x1** [comment]
@palisadoes added git hook

**Comment by palisadoes** [comment]
I thought about the universality of this solution. Will it work for developers running on the most OS  platforms?

1. Windows
1. Mac
1. Linux

For example, the find command is Unix specific

**Comment by Veer0x1** [comment]
> I thought about the universality of this solution. Will it work for developers running on the most OS platforms?
> 
> 1. Windows
> 2. Mac
> 3. Linux
> 
> For example, the find command is Unix specific

It's my bad; I forgot to consider Windows; the `find` command will work for Unix-based Operating systems only, so to make it more universal, we could write a TypeScript file, which will find the markdown files using Node.js's files system. Or if you have any other approach, please suggest.

**Comment by palisadoes** [comment]
1. Yes typescript works. 
1. Please the script in the `scripts`directory. Under a dedicated `githooks`sub directory to make it very obvious.
1. Move any other scripts used by hooks there too

**Comment by palisadoes** [comment]
We have a similar PR open in Admin. Javascript was selected for the reason stated here
- https://github.com/PalisadoesFoundation/talawa-admin/pull/1280#issuecomment-1871411776

**Comment by Veer0x1** [comment]
> We have a similar PR open in Admin. Javascript was selected for the reason stated here
> 
> * [Feature :  Added markdown-toc package to auto-generate toc with husky pre commit hook. talawa-admin#1280 (comment)](https://github.com/PalisadoesFoundation/talawa-admin/pull/1280#issuecomment-1871411776)

Thanks for suggesting this, I have written the script, I just wanted to ask do you want Table of Content for README.md file?

**Comment by palisadoes** [comment]
No that's not necessary

**Comment by palisadoes** [comment]
Yes, let's keep it for consistency

**Review by palisadoes — APPROVED** [review]

**Pull request merged**